src/main/java/redis/clients/jedis/ShardedJedis.java

```diff
@@ -785,21 +785,14 @@ public Long bitpos(String key, boolean value, BitPosParams params) {
   @Override
   public void close() {
     if (dataSource != null) {
-      boolean broken = false;
-
-      for (Jedis jedis : getAllShards()) {
+      for (JedisShardInfo shardInfo: getDistinctShardInfo()) {
+        Jedis jedis = getShard(shardInfo);
         if (jedis.getClient().isBroken()) {
-          broken = true;
-          break;
+          replaceShard(shardInfo);
+          Jedis.destroy(jedis);
```

Comment:
Do we really need to force `destroy` if the connection is broken?. I think the `quit` call will always fail in the `destroy` function if the connection is broken.